src/MessageEditor.js

```diff
@@ -14,23 +21,69 @@ class MessageEditor extends React.Component {
 
   componentDidUpdate(prevProps) {
     if (this.props.id !== prevProps.id && this.props.id) {
+      this.setState({
+        messages: [],
+        gauges: []
+      });
       api.fetch(`glyphs/${this.props.id}/messages`)
       .then(messages => this.setState({ messages: messages }));
 
       api.fetch(`glyphs/${this.props.id}/gauges`)
       .then(gauges => this.setState({ gauges: gauges }));
+
     }
   }
 
   render() {
-    const { messages, gauges } = this.state;
+    let { messages, gauges } = this.state;
+
+    const addVMessage = () => {
+      api.post(`glyphs/${this.props.id}/messages/`, {"pass" : this.inputRef.value, "text" : 'Default message', "probability" : 0}).then(result => {console.log(result)})
+      messages.push({"text" : 'Default message', "probability" : 0})
+      this.forceUpdate();
```

Comment:
This code is anti-React, actually.  The correct way to do what you are trying to do is keeping state `const` and using `this.setState`.  You would write something like `this.setState({ messages: [...messages, { what you push }] })`